### DISCUSSION TITLE
Exporting data

### DISCUSSION THREAD

**Discussion by SharmaAntriksh** [DaxStudio/DaxStudio#842] [Q&A] [answered]
Just curious to know what exactly happens behind the scenes when we try to export data to CSV for a Database, is DAX Studio running some SSIS package or is it a DAX expression like EVALUATE Table?

**Comment by dgosbell** [comment, marked as answer]
I just run an `EVALUATE table` expression over each table in the model. However if the table contains more than 1 million records I run a series of EVALUATE statements using the TOPNSKIP function to grab 1 million rows at a time.  This prevents the tabular engine from having to decompress and materialize a copy of the table in memory which helps avoid out of memory issues in the tabular engine. I also stream the results to the output (SQL tables or CSV files) as they arrive so that I do not store the entire result in memory in DAX Studio.
